src/shared/utils/sqlInterval.ts

```diff
@@ -0,0 +1,88 @@
+import {TimeRange} from 'src/types'
+import {DEFAULT_TIME_RANGE} from 'src/shared/constants/timeRanges'
+
+const durationExpr = /^\-(([0-9]+)(y|mo|w|d|h|ms|s|m|us|µs|ns))+$/g
+const singleDuration = /(([0-9]+)(y|mo|w|d|h|ms|s|m|us|µs|ns))/g
+const nanosecondDuration = /(([0-9]+)(ns))/g
+const microsecondDuration = /(([0-9,\.]+)(\W?)(microsecond))/g
+
+const convertToInterval = (upperOrLower: string) => {
+  if (upperOrLower == null || upperOrLower == 'now()') {
+    return 'now()'
+  }
+  if (!!upperOrLower.match(durationExpr)) {
+    let interval = [
+      ...upperOrLower
+        .replace('-', '') // -12d6h => 12d6h
+        .matchAll(singleDuration), // [['12d','12','h'],['6h','6','h']]
+    ]
+      .map(x => x[0]) // because want most greedy match for each
+      .join(' ') // '12d 6h'
+      .replace('y', ' year')
+      .replace('h', ' hour') // match h, before month with h
+      .replace('mo', ' month')
+      .replace('w', ' week')
+      .replace('d', ' day')
+      .replace('ms', ' millisecond') // match ms, before m|s
+      .replace('us', ' microsecond')
+      .replace('µs', ' microsecond')
+      .replace('m ', ' minute ') // don't match `m` in millisecond, etc
+      .replace('s ', ' second ')
+
+    // postgres does not support nanoseconds. Convert.
+    let toReplace = interval.match(nanosecondDuration)
+    if (!!toReplace || toReplace?.length > 0) {
+      try {
+        toReplace.forEach(init => {
+          const replacement = `${
+            parseFloat(init.split('ns')[0]) / 1000
+          } microsecond`
+          interval = interval.replace(init, replacement)
+        })
+      } catch (e) {
+        throw new Error(`Cannot handle nanosecond: ${e}`)
+      }
+    }
+
+    // our flight sql is not supporting microseconds. Convert.
+    toReplace = interval.match(microsecondDuration)
+    if (!!toReplace || toReplace?.length > 0) {
```

Comment:
Evidence of error:
<img width="589" alt="Screen Shot 2023-01-04 at 2 37 03 PM" src="https://user-images.githubusercontent.com/10232835/210682112-fc5e5ce2-f552-4d89-b8ee-744dcad3357b.png">